pandas/core/internals/managers.py

```diff
@@ -968,19 +968,10 @@ def fast_xs(self, loc: int) -> SingleBlockManager:
 
         n = len(self)
 
-        # GH#46406
-        immutable_ea = isinstance(dtype, ExtensionDtype) and dtype._is_immutable
-
-        if isinstance(dtype, ExtensionDtype) and not immutable_ea:
-            cls = dtype.construct_array_type()
-            result = cls._empty((n,), dtype=dtype)
+        if isinstance(dtype, ExtensionDtype):
+            result = np.empty(n, dtype=object)
```

Comment:
i expect this will be bad for e.g. DatetimeTZDtype

Reply:
I just tried it for `DatetimeTZDtype("ns", "UTC")` and it seems to be ok - and more performant in that case as well. 

I think it's ok since each element is pulled out individually which ensures wrapping in `Timestamp`.